src/SpecFlow.NetCore/AppConfig.cs

```diff
@@ -18,59 +21,117 @@ internal class AppConfig
 		public const string SpecFlowSectionElement = "unitTestProvider";
 
 		public const string SpecFlowSection = @"	<specFlow>
-		<" + SpecFlowSectionElement + @" name=""xUnit"" />
+		<" + SpecFlowSectionElement + @" name=""{0}"" />
 	</specFlow>";
 
 		public static string Content = $@"<?xml version=""1.0"" encoding=""utf-8""?>
 <configuration>
 {SpecFlowSectionDefinition}
 {SpecFlowSection}
-</configuration>";
+</configuration>";
 		#endregion
-
+
 		public string Path { get; }
 
-		public AppConfig(string path)
+		public string TestFramework { get; }
+
+		public AppConfig(string path, string testFramework)
 		{
 			Path = path;
+			TestFramework = testFramework;
 		}
 
-		public static AppConfig CreateIn(DirectoryInfo directory)
+		public static AppConfig CreateIn(DirectoryInfo directory, string testRunner = null)
 		{
-			var config = new AppConfig(System.IO.Path.Combine(directory.FullName, "app.config"));
+			if (string.IsNullOrWhiteSpace(testRunner))
+			{
+				testRunner = GuessUnitTestProvider(directory);
+			}
+			var config = new AppConfig(System.IO.Path.Combine(directory.FullName, "app.config"), testRunner);
 
+			WriteLine($@"Using test runner ""{testRunner}""");
 			if (File.Exists(config.Path))
+			{
+				config.ChangeTestProviderIfNeeded(testRunner);
 				return config;
+			}
 
-			WriteLine("Generating app.config.");
+			Content = string.Format(Content, testRunner);
+			WriteLine("Generating app.config");
 			WriteLine(Content);
 			WriteLine("Saving: " + config.Path);
 			File.WriteAllText(config.Path, Content);
 
 			return config;
 		}
 
-		public void Validate()
+		public void ChangeTestProviderIfNeeded(string testRunner)
```

Comment:
This is a pretty wordy and brittle method (if it doesn't find the exact app config stuff it does nothing silently). It also means it is no longer possible to change the app.config testRunner as we will overwrite it. Probably not a big deal as test runner is now specified using the arguments instead.